DependencyLibrary.py

```diff
@@ -22,6 +22,8 @@ def _depends_on(status_map, dependency_type, name):
         return
     if status == 'SKIP':
         raise _SkipExecution(message + ' was skipped.')
+    if skip:
+        raise _SkipExecution(message + ' was skipped because dependency failed.')
```

Comment:
Now that I'm actually looking at this closely in a brain-on way, I'm realizing that the correct skip message ending is just `' failed.'`

We already know *this test* is being skipped because the status is skip, the message is telling us *why* this status is happening: in this case, this test is skipping because the dependency failed.